src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseTestService.java

```diff
@@ -1701,13 +1701,13 @@ void copyRepository_testNotCreatedError() throws Exception {
         doThrow(new CanceledException("Checkout got interrupted!")).when(gitService).getOrCheckoutRepositoryIntoTargetDirectory(any(), any(), anyBoolean());
 
         // the local repo should exist before startExercise()
-        assertThat(Files.exists(teamLocalPath)).isTrue();
+        assertThat(teamLocalPath).exists();
         // Start participation
-        var exception = assertThrows(VersionControlException.class, () -> participationService.startExercise(exercise, team, false));
+        assertThatExceptionOfType(VersionControlException.class).isThrownBy(() -> participationService.startExercise(exercise, team, false))
+                .matches(exception -> !exception.getMessage().isEmpty());
         // the directory of the repo should be deleted
-        assertThat(Files.exists(teamLocalPath)).isFalse();
+        assertThat(teamLocalPath).doesNotExist();
         // We cannot compare exception messages because each vcs has their own. Maybe simply checking that the exception is not empty is enough?
```

Comment:
```suggestion
```
Or move up. However, "maybe" comments are usually not really worth keeping

Reply:
I decided to remove it :+1: